tests/commands/patch.py

```diff
@@ -3,60 +3,78 @@
 """
 
 
-from tests.utils import debug_target, gdb_run_cmd, gdb_run_silent_cmd, gdb_start_silent_cmd_last_line
-from tests.utils import GefUnitTestGeneric
+from tests.base import RemoteGefUnitTestGeneric
+from tests.utils import ERROR_INACTIVE_SESSION_MESSAGE, debug_target, u16, u32, u64, u8
 
 
-class PatchCommand(GefUnitTestGeneric):
+class PatchCommand(RemoteGefUnitTestGeneric):
     """`patch` command test module"""
 
-
     def test_cmd_patch(self):
-        self.assertFailIfInactiveSession(gdb_run_cmd("patch"))
-
+        gdb = self._gdb
+        self.assertEqual(
+            ERROR_INACTIVE_SESSION_MESSAGE, gdb.execute("patch", to_string=True)
+        )
 
     def test_cmd_patch_byte(self):
-        res = gdb_start_silent_cmd_last_line("patch byte $pc 0xcc", after=["display/8bx $pc",])
-        self.assertNoException(res)
-        self.assertRegex(res, r"0xcc\s*0x[^c]{2}")
+        gdb = self._gdb
+        gef = self._gef
+        gdb.execute("start")
+        gdb.execute("patch byte $pc 0xcc")
+        mem = u8(gef.memory.read(gef.arch.pc, 1))
+        self.assertEqual(mem, 0xCC)
 
     def test_cmd_patch_byte_bytearray(self):
-        res = gdb_start_silent_cmd_last_line("set $_gef69 = { 0xcc, 0xcc }", after=["patch byte $pc $_gef69", "display/8bx $pc",])
-        self.assertNoException(res)
-        self.assertRegex(res, r"(0xcc\s*)(\1)0x[^c]{2}")
+        gdb = self._gdb
+        gef = self._gef
+        gdb.execute("start")
+        gdb.execute("set $_gef69 = { 0xcc, 0xcc }")
+        gdb.execute("patch byte $pc $_gef69")
+        mem = u16(gef.memory.read(gef.arch.pc, 2))
+        self.assertEqual(mem, 0xCCCC)
 
     def test_cmd_patch_word(self):
-        res = gdb_start_silent_cmd_last_line("patch word $pc 0xcccc", after=["display/8bx $pc",])
-        self.assertNoException(res)
-        self.assertRegex(res, r"(0xcc\s*)(\1)0x[^c]{2}")
-
+        gdb = self._gdb
+        gef = self._gef
+        gdb.execute("start")
+        gdb.execute("patch word $pc 0xcccc")
+        mem = u16(gef.memory.read(gef.arch.pc, 2))
+        self.assertEqual(mem, 0xCCCC)
 
     def test_cmd_patch_dword(self):
-        res = gdb_start_silent_cmd_last_line("patch dword $pc 0xcccccccc",
-                                             after=["display/8bx $pc",])
-        self.assertNoException(res)
-        self.assertRegex(res, r"(0xcc\s*)(\1\1\1)0x[^c]{2}")
-
+        gdb = self._gdb
+        gef = self._gef
+        gdb.execute("start")
+        gdb.execute("patch dword $pc 0xcccccccc")
+        mem = u32(gef.memory.read(gef.arch.pc, 4))
+        self.assertEqual(mem, 0xCCCCCCCC)
 
     def test_cmd_patch_qword(self):
-        res = gdb_start_silent_cmd_last_line("patch qword $pc 0xcccccccccccccccc",
-                                             after=["display/8bx $pc",])
-        self.assertNoException(res)
-        self.assertRegex(res, r"(0xcc\s*)(\1\1\1\1\1\1)0xcc")
```

Comment:
Nice work getting rid of these regexes!

Reply:
IMHO tests are much more readable as-is so I'm very happy.
I hope it'll help testing more, and increasing coverage.